Feat: Pages and Simple Sections

thapacodes4u opened the pull request:
[Related Ticket](https://datalinx.atlassian.net/browse/CM-8)

Commit:
feat: adding pages and sections resource with test and other things

Commit:
reverting User workbench

Commit:
fix: remove unnecessary enums sections

SlimDeluxe reviewed (changes requested):
Please see below comments before I move to the UI test.

SlimDeluxe commented on database/migrations/2025_07_22_082135_create_sections_table.php:
```diff
@@ -14,12 +14,15 @@ public function up(): void
+                        ->cascadeOnUpdate()
+                        ->cascadeOnDelete();
+                } else {
+                    $table->unsignedBigInteger(config('eclipse-cms.tenancy.foreign_key'))->nullable();
```
Yes, but only when used in a multi-tenant app.
For other, the tenancy config isn't set, so the migration with your change would not work. So, for this case, the FK column cannot be created.

SlimDeluxe commented on src/Models/Page.php:
```diff
@@ -25,13 +33,117 @@ class Page extends Model
         return $this->belongsTo(Section::class);
     }
 
+    public function site()
+    {
+        return $this->hasOneThrough(
+            config('eclipse-cms.tenancy.model'),
+            Section::class,
+            'id',
+            'id',
+            'section_id',
+            'site_id'
+        );
+    }
```
When tenancy is used, we have the FK in the section, and the section has pages. Pages do not have or need the tenant FK. So your relation in Page class is OK I guess, but I was wondering why is it needed. Can it be renamed to tenant(), to make the plugin compatible with other setups/terminology?

Commit:
refactor: implement feedback and clean up tests

Commit:
refactor: remove redundant code and standardize test naming

Commit:
refactor: remove redundant code and standardize test naming

SlimDeluxe reviewed (changes requested):
1. There's an error when creating a new section

<img width="1147" height="351" alt="Image" src="https://github.com/user-attachments/assets/6fb705ff-32f2-476a-b076-e978a8bed9f6" />

2. Sections are not scoped to tenant? I can see the same sections on both sites.
3. I don't know how but you missed the instruction to make sections as nav menu items. I even included a screenshot from the old app. But you checked as Done anyway.

<img width="459" height="30" alt="Image" src="https://github.com/user-attachments/assets/db23aa17-dcdd-4524-8b5d-30b342180027" />

4. Same goes for tenant test, marked as Done but I can't see it.

SlimDeluxe commented:
Before continuing please merge from the main branch, I updated the CI workflows.

Commit:
Merge branch 'main' of github.com:thapacodes4u/eclipsephp-cms-plugin into feat/pages-and-sections

SlimDeluxe commented:
@thapacodes4u your last commit in this PR was on 11th Aug, but I made a change to the CI files 2 days later. If you merge from main, you'll get the fixed workflows so that tests in the PR work.

Commit:
feat: implement pages and sections with tenant scoping & navigation

Commit:
Refactor: Using right URL for dynamic section & Page filtering accordingly & test cases

Commit:
Refactor: renaming the query string & also adding it to the create action

Commit:
Fixing: SQL error when creating new page

SlimDeluxe reviewed (changes requested):
- [x] Remove Pages link from nav

<img width="278" height="197" alt="Image" src="https://github.com/user-attachments/assets/1cf3ebb1-1dde-4207-a0ff-61bf50497ab6" />

- [x] Dynamic section list should be before the Sections link and use the icon that is currently set for Pages

<img width="270" height="185" alt="Image" src="https://github.com/user-attachments/assets/ae7af0a7-956e-436c-84cf-9d115a976f97" />

- [x] Pages management should be it's own page. It should not be done in the section. Some editors can have only right to create/update pages, but not sections, so these must clearly be separated. So please remove the page management from the section view and edit pages, and make it fully manageable on the dynamic page.
- [x] Improve the dynamic page: 
a) Use the section name for the page title and breadcrumb, 
b) Remove the Section column

<img width="1261" height="424" alt="Image" src="https://github.com/user-attachments/assets/17e699b5-8fd6-4eba-b63e-994bc8619d78" />

- [x] When I toggle some columns, the refreshed table shows pages from other sections too. E.g. adding the column code, table refreshes and shows all pages, even from other sections, but then when you reload the page, again the section filter is correctly applied

Commit:
feat: separate pages and sections management with improved navigation

-[x] Remove Pages link from navigation menu
-[x] Reorder dynamic section links before Sections link with document icon
-[x] Remove page management from section view (deleted PagesRelationManager)
-[x] Use section name for page title and breadcrumb on dynamic pages
-[x] Hide Section column when viewing pages filtered by section
-[x] Fix table filtering bug by moving section filter to getTableQuery

Commit:
refactor: improve seeder and remove unnecessary view permissions
  - Add tenant-aware seeding for sections and pages
  - Create 5 sections with 2-5 random pages per tenant
  - Fix page code uniqueness using numeric pattern instead of words
  - Remove individual view permissions from PagePolicy and SectionPolicy
  - Remove view action from section edit page
  - Remove view from permission prefixes in resources

Commit:
fix: section filter now persists when toggling columns

SlimDeluxe reviewed (changes requested):
Again some conflicts appeared, so please update your PR

Commit:
Resolving conflict

SlimDeluxe reviewed (changes requested):
There are some conflicts to resolve here.

thapacodes4u committed:
fix: resolving conflict & fixing failing tests

thapacodes4u committed:
fix: failing test

thapacodes4u commented:
@SlimDeluxe Should be ready for review

SlimDeluxe reviewed (changes requested):
1. The seeder is generating too many sections per site (10). Let's do 2 per site instead.
2. Short description (in page editing) should also be a WYS editor.
3. When editing an existing page, you can't clear the "URL slug" field so that it'll be auto-generated from the new page title. E.g.:
- edit page, change title to "abc"
- the slug is not changed, which is correct
- but also when you clear the slug and try to save, you can't, since the slug field is required, but also it does not auto-generate

thapacodes4u committed:
fix: reduce section seeding, add WYSIWYG to short description, allow slug auto-generation

thapacodes4u committed:
Merge branch 'main' into feat/pages-and-sections

SlimDeluxe committed:
refactor: update translatable property format in Section model
